### PR TITLE
Fix noobaa label issue for upgrade to OCS 4.7

### PR DESCRIPTION
**Pull request by petr-balogh**
Fixes: #4068

Signed-off-by: Petr Balogh <pbalogh@redhat.com>

### CODE REVIEW

**Comment by petr-balogh** [comment]
Verification triggered here:
https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/view/Upgrade-OCS/job/qe-trigger-vsphere-upi-1az-rhcos-vsan-3m-3w-upgrade-ocs-auto/32/console

File: ocs_ci/ocs/ocs_upgrade.py [hunk 1]
```diff
@@ -90,10 +90,34 @@ def verify_image_versions(old_images, upgrade_version, version_before_upgrade):
...
+        )
+    except TimeoutException as ex:
+        log.info(
+            "Nooba pods didn't match. Trying once more with max noobaa endpoints!"
+            f"Exception: {ex}"
+        )
+        verify_pods_upgraded(
+            old_images,
+            selector=constants.NOOBAA_APP_LABEL,
+            count=noobaa_pods + max_endpoints,
```
Comment (dahorak) [inline ocs_ci/ocs/ocs_upgrade.py:118]: Same here with `max_endpoints`.

**Review by dahorak — CHANGES_REQUESTED** [review]

File: ocs_ci/ocs/ocs_upgrade.py [hunk 2]
```diff
@@ -90,10 +90,34 @@ def verify_image_versions(old_images, upgrade_version, version_before_upgrade):
...
+        endpoints = resource.get("spec", {}).get("endpoints", {})
+        max_endpoints = endpoints.get("maxCount", constants.MAX_NB_ENDPOINT_COUNT)
+        min_endpoints = endpoints.get(
+            "minCount", constants.MIN_NB_ENDPOINT_COUNT_POST_DEPLOYMENT
+        )
+    try:
+        verify_pods_upgraded(
+            old_images,
+            selector=constants.NOOBAA_APP_LABEL,
+            count=noobaa_pods + min_endpoints,
```
[the remark this answers is not in the document]
Comment (ebenahar) [reply ocs_ci/ocs/ocs_upgrade.py:107]: @dahorak `min_endpoints` and `max_endpoints` exist in the nooaa CR starting OCS 4.6

File: ocs_ci/ocs/ocs_upgrade.py (hunk 1 above)
Comment (petr-balogh) [reply ocs_ci/ocs/ocs_upgrade.py]: Good catch Daniel, I  sent a fix.

File: ocs_ci/ocs/ocs_upgrade.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dahorak) [reply ocs_ci/ocs/ocs_upgrade.py]: My concern was about python variables not defined under the condition mentioned above - fixed in the recent update.

**Review by dahorak — DISMISSED** [review]

**Comment by petr-balogh** [comment]
New verification triggered here: https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/view/Upgrade-OCS/job/qe-trigger-aws-ipi-3az-rhcos-3m-3w-upgrade-ocs-auto/10/console

File: ocs_ci/ocs/ocs_upgrade.py [hunk 3]
```diff
@@ -90,10 +90,34 @@ def verify_image_versions(old_images, upgrade_version, version_before_upgrade):
     verify_pods_upgraded(old_images, selector=constants.OPERATOR_LABEL)
     # in 4.3 app selector nooba have those pods: noobaa-core-ID, noobaa-db-ID,
     # noobaa-operator-ID but in 4.2 only 2: noobaa-core-ID, noobaa-operator-ID
-    nooba_pods = 2 if upgrade_version < parse_version("4.3") else 3
-    verify_pods_upgraded(
-        old_images, selector=constants.NOOBAA_APP_LABEL, count=nooba_pods
-    )
+    noobaa_timeout = 1200 if upgrade_version == parse_version("4.7") else 720
+    noobaa_pods = 2 if upgrade_version < parse_version("4.3") else 3
```
[the remark this answers is not in the document]
Comment (petr-balogh) [reply ocs_ci/ocs/ocs_upgrade.py]: Missed this comment. I am running verification now how it is..  Next time I will touch this part I can change.

**Review by jimdevops19 — APPROVED** [review]

**Review by ebenahar — DISMISSED** [review]

**Review by dahorak — APPROVED** [review]

File: ocs_ci/ocs/ocs_upgrade.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (petr-balogh) [reply ocs_ci/ocs/ocs_upgrade.py]: Finally I removed this as part of this PR when I did last changes

**Pull request merged**